File: app/src/main/java/com/app/applemvp/MainActivity.java
package com.app.applemvp;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.app.applemvp.mvp.MainPresenter;
import com.app.applemvp.mvp.SplashPresenter;
import com.app.applemvp.mvp.SplashView;
import com.base.applemvp.annotations.CreatePresenterAnnotation;
import com.base.applemvp.common.BaseMvpActivity;
import com.app.applemvp.mvp.MainView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;


public class MainActivity extends BaseMvpActivity implements SplashView, MainView{


    @BindView(R.id.btnSync)
    Button btnSync;
    @BindView(R.id.btnAdvert)
    Button btnAdvert;


    @CreatePresenterAnnotation(SplashPresenter.class)
    SplashPresenter splashPresenter;

    @CreatePresenterAnnotation(MainPresenter.class)
    MainPresenter mainPresenter;

    @Override
    public int setLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    protected void initData() {

    }

    @Override
    protected void initView(Bundle savedInstanceState) {
        ButterKnife.bind(this);
    }

    @Override
    public void appSync(String msg) {
        btnSync.setText(msg);
    }

    @Override
    public void advertList(String advert) {
        btnAdvert.setText(advert);
    }


    @OnClick({R.id.btnSync, R.id.btnAdvert})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.btnSync:
                mainPresenter.appSync("首页初始化");
                break;
            case R.id.btnAdvert:
                splashPresenter.advertList("启动页初始化");
                break;
        }
    }
}

File: app/src/main/java/com/app/applemvp/MainActivity.kt
package com.app.applemvp

import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.databinding.DataBindingUtil
import com.app.applemvp.databinding.ActivityMainBinding
import com.app.applemvp.mvp.MainPresenter
import com.app.applemvp.mvp.MainView
import com.base.applemvp.annotations.CreatePresenterAnnotation
import com.base.applemvp.common.BaseMvpActivity


class MainActivity : BaseMvpActivity(), MainView {

    lateinit var binding: ActivityMainBinding
    lateinit var entity:MainEntity
    @CreatePresenterAnnotation(MainPresenter::class)
    var mainPresenter: MainPresenter? = null


    override fun initData() {
        Log.e("HU", "initData======" + mainPresenter);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.lifecycleOwner = this
        entity= MainEntity(application)
        binding.userInfo = entity
        binding.mainAct=this
        mainPresenter?.appSync("xxxxxxx")
    }

    override fun initView(savedInstanceState: Bundle?) {

    }

    override fun appSync(msg: String) {
        Log.e("HU", "appSync===xx===" + msg);
        entity.name.postValue(msg)
    }


    fun  onClick(view:View){
        Log.e("HU", "onClick")
        entity.name.postValue("mmmmm"+view.id.toString())
    }


}